storm-core/src/jvm/org/apache/storm/trident/windowing/AbstractTridentWindowManager.java

```diff
@@ -0,0 +1,241 @@
+
+        // run aggregator to compute the result
+        AccumulatedTuplesCollector collector = new AccumulatedTuplesCollector(delegateCollector);
+        Object state = aggregator.init(currentTriggerId, collector);
+        for (TridentTuple resultTuple : resultTuples) {
+            aggregator.aggregate(state, resultTuple, collector);
+        }
+        aggregator.complete(state, collector);
+
+        List<List<Object>> resultantAggregatedValue = collector.values;
+
+        ArrayList<WindowsStore.Entry> entries = Lists.newArrayList(new WindowsStore.Entry(windowTriggerCountId, currentTriggerId + 1),
+                new WindowsStore.Entry(WindowTridentProcessor.generateWindowTriggerKey(windowTaskId, currentTriggerId), resultantAggregatedValue));
+        windowStore.putAll(entries);
+
+        pendingTriggers.add(new TriggerResult(currentTriggerId, resultantAggregatedValue));
+    }
+
+    /**
+     * Return {@code TridentTuple}s from given {@code tupleEvents}.
+     * @param tupleEvents
+     * @return
+     */
+    protected abstract List<TridentTuple> getTridentTuples(List<T> tupleEvents);
+
+    /**
+     * This {@code TridentCollector} accumulates all the values emitted.
+     */
+    static class AccumulatedTuplesCollector implements TridentCollector {
+
+        final List<List<Object>> values = new ArrayList<>();
+        private final BatchOutputCollector delegateCollector;
+
+        public AccumulatedTuplesCollector(BatchOutputCollector delegateCollector) {
+            this.delegateCollector = delegateCollector;
+        }
+
+        @Override
+        public void emit(List<Object> values) {
+            this.values.add(values);
+        }
+
+        @Override
+        public void reportError(Throwable t) {
+            delegateCollector.reportError(t);
+        }
+
+    }
+
+    static class TriggerResult {
+        final int id;
+        final List<List<Object>> result;
+
+        public TriggerResult(int id, List<List<Object>> result) {
+            this.id = id;
+            this.result = result;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof TriggerResult)) return false;
+
+            TriggerResult that = (TriggerResult) o;
+
+            return id == that.id;
+
+        }
+
+        @Override
+        public int hashCode() {
+            return id;
+        }
+
+        @Override
+        public String toString() {
+            return "TriggerResult{" +
+                    "id=" + id +
+                    ", result=" + result +
+                    '}';
+        }
+    }
+
+    public Queue<TriggerResult> getPendingTriggers() {
+        return pendingTriggers;
+    }
+
+    public void shutdown() {
+        try {
+            log.info("window manager [{}] is being shutdown", windowManager);
+            windowManager.shutdown();
+        } finally {
```

Comment:
I will add logging an exception in WindowTridentProcessor from which this method is invoked.